omnigibson/simulator.py

```diff
@@ -569,6 +591,48 @@ def add_skybox(self):
             self._skybox.color = (1.07, 0.85, 0.61)
             self._skybox.texture_file_path = f"{gm.ASSET_PATH}/models/background/sky.jpg"
 
+        def get_sim_step_dt(self):
+            """
+            Gets the internal simulation step timestep size
+
+            Returns:
+                float: Simulation timestep size
+            """
+            return self._sim_step_dt
+
+        def set_sim_step_dt(self, sim_step_dt):
+            """
+            Sets the internal simulation step timestep size
+
+            Args:
+                sim_step_dt (float): Timestep size to apply
+            """
+            # Sanity check this new dt with respect to the rendering dt
+            rendering_dt = self.get_rendering_dt()
+            sim_render_ratio = sim_step_dt / rendering_dt
+            assert np.isclose(sim_render_ratio, np.round(sim_render_ratio)), \
+                f"Simulation step dt ({sim_step_dt}) must be a multiple of rendering dt ({rendering_dt})"
+            assert sim_step_dt > rendering_dt, \
+                f"Simulation step dt ({sim_step_dt}) cannot be smaller than rendering dt ({rendering_dt})"
+            self._sim_step_dt = sim_step_dt
+            self._n_steps_per_loop = int(sim_step_dt // rendering_dt)
+
+        def set_simulation_dt(self, physics_dt=None, rendering_dt=None):
+            # Call super first
+            super().set_simulation_dt(physics_dt=physics_dt, rendering_dt=rendering_dt)
+
+            # Sanity check all new dts
+            rendering_dt = self.get_rendering_dt()
+            physics_dt = self.get_physics_dt()
+            render_physics_ratio = rendering_dt / physics_dt
+            sim_render_ratio = self._sim_step_dt / rendering_dt
+            assert np.isclose(render_physics_ratio, np.round(render_physics_ratio)), \
+                f"Rendering dt ({rendering_dt}) must be a multiple of physics dt ({physics_dt})"
+            assert np.isclose(sim_render_ratio, np.round(sim_render_ratio)), \
+                f"Simulation step dt ({self._sim_step_dt}) must be a multiple of rendering dt ({rendering_dt})"
+            assert self._sim_step_dt > rendering_dt, \
+                f"Simulation step dt ({self._sim_step_dt}) cannot be smaller than rendering dt ({rendering_dt})"
```

Comment:
should we put these sanity check in a helper function and reuse?

Reply:
done